ExpressMapper NET40/MappingServiceBase.cs

```diff
@@ -423,28 +412,56 @@ protected LoopExpression CreateLoopExpression(Expression doMoveNextSrc, BlockExp
             {
                 return Expression.Call(destColl, destList.GetMethod("ToArray"));
             }
-            if (!destPropType.IsGenericType) return destColl;
+
+            if (!destPropType.IsGenericType && GetEnumerableInterface(destPropType) == null)
+            {
+                return destColl;
+            }
 
             if (typeof(IQueryable).IsAssignableFrom(destPropType))
             {
                 return Expression.Call(typeof(Queryable), "AsQueryable", new[] { destType }, destColl);
             }
 
-            if (destPropType.IsInterface && destColl.Type.IsSubclassOf(destPropType))
-            {
+            if (destPropType.IsInterface && destPropType.IsAssignableFrom(destColl.Type)) {
+                // This will handle any destination interface implemented by List<T>

```

Comment:
From https://msdn.microsoft.com/en-us/library/system.type.issubclassof(v=vs.110).aspx :
"The IsSubclassOf method cannot be used to determine whether an interface derives from another interface, or whether a class implements an interface. Use theIsAssignableFrom method for that purpose, as the following example shows."

Since we create destColl as List<>, this will handle all cases where the target type is an interface and that interface is implemented by list.